docs/rules/forbid-elements.md

```diff
@@ -0,0 +1,39 @@
+# Forbid certain elements (forbid-elements)
+
+You may want to forbid usage of certain elements in favor of others, (e.g. forbid all `<div />` and use `<Box />` instead). This rule allows you to configure a list of forbidden elements and to specify their desired replacements.
+
+## Rule Details
+
+This rule checks all JSX components and verifies that no forbidden elements are used. When it detects a forbidden element, the error message will contain the specified elements that you should use instead. This rule is off by default. If on, no elements are forbidden by default.
```

Comment:
Sorry, made some typos earlier. I guess I am arguing for two things:

1) If 'button' is forbidden, I don't think this rule should stop someone from doing `const button = 'div'; React.createElement(button)` because that looks like that's done for a reason other than to circumvent the rule (otherwise they would just use `eslint-ignore`).

2) In general:
* if `'button'` is forbidden, then the rule should check for `React.createElement('button')` and ignore  `React.createElement(button)`
* if `'Modal'` is forbidden, then the rule should check for `React.createElement(Modal)` and ignore `React.createElement('Modal')`.
* if `'a.b'` is forbidden, then the rule should check for `React.createElement(a.b)` and ignore `React.createElement('a.b')`.

because this aligns with how React translates JSX element names into createElement calls